### PR TITLE
Capture service worker intercepted requests

### PR DESCRIPTION
**Pull request by nilshah98**
**Changes**:
- [x] Added `captureMockedServiceWorker` flag under `discovery` for capturing service worker requests
- [x]  Set `bypass: false` for `Network.setBypassServiceWorker`  so that service-worker controller can be added and we can access the MSW mocks customer has added.
- [x] Added `serviceWorker` parameter, to handleRequests that are not intercepted _(since service worker intercepted and will resolve them)_

**Detailed Changes & Reasoning**:
- Added `Network.setBypassServiceWorker` to `bypass: false` so that we can add service-worker controller and access service worker that mocks customer's data.
- Initially tried to access service-worker requests using `Network.getResponseBody`, but over here `body` is returned as `""` _(empty string)_ randomly.
- - Tried to look on the internet about why this happens, but no luck :(
- Finally, decided to call `Network.getResponseBody` on the original request which service-worker intercepted.
- - `Network.getResponseBody` is same as what appears in the `Response` tab in Network section in devtools.
- - For requests that were intercepted by service-worker, the `Response` tab is **always** populated.
- - But the service-worker which makes the request upstream or returns mocked data randomly returns `body: ""` when `Network.getResponseBody` is called on it.
- - Hence, relying on calling `Network.getResponseBody` on the original request itself.

**Related PRs**:
https://github.com/percy/percy-storybook/pull/845

### CODE REVIEW

File: packages/core/src/discovery.js [hunk 1]
```diff
@@ -288,6 +289,7 @@ export function createDiscoveryQueue(percy) {
         requestHeaders: snapshot.discovery.requestHeaders,
         authorization: snapshot.discovery.authorization,
         userAgent: snapshot.discovery.userAgent,
+        captureServiceWorker: snapshot.discovery.captureServiceWorker,
```
[the remark this answers is not in the document]
Comment (itsjwala) [reply packages/core/src/discovery.js:292]: `snapshot.discovery.captureServiceWorker` will contain global value if, can you verify once?

File: packages/core/src/snapshot.js [hunk 2]
```diff
@@ -114,6 +114,7 @@ function getSnapshotOptions(options, { config, meta }) {
       requestHeaders: config.discovery.requestHeaders,
       authorization: config.discovery.authorization,
       disableCache: config.discovery.disableCache,
+      captureServiceWorker: config.discovery.captureServiceWorker,
```
Comment (itsjwala) [inline packages/core/src/snapshot.js:117]: okay we're overriding it here so no issues.

**Review by itsjwala — COMMENTED** [review]
can you link locking CLI pr?

**Comment by nilshah98** [comment]
> can you link locking CLI pr?

The base branch for this one is https://github.com/percy/cli/tree/refactor-network-js.
Hence, for merging as well, the PR is created for merging into https://github.com/percy/cli/tree/refactor-network-js.

File: packages/core/test/discovery.test.js [hunk 3]
```diff
@@ -1970,4 +1970,68 @@ describe('Discovery', () => {
...
+        event.waitUntil(clients.claim());
+      });
+      `]);
+
+      server.reply('/app.js', () => [200, 'text/javascript', dedent`
+      const registerServiceWorker = async () => {
+        await navigator.serviceWorker.register('sw.js',{ scope: './', });
+      };
+      
+      await registerServiceWorker();
```
Comment (ninadbstack) [inline packages/core/test/discovery.test.js:1996]: didnt know we can call await in base js context

Comment (nilshah98) [reply packages/core/test/discovery.test.js:1996]: The script declaration is done as -
```html
    <script type="module" src="app.js"></script>
```

Here, `type="module"` allows for using `await` at top-level.
If we remove `type="module"` it throws error as -

`Uncaught SyntaxError: await is only valid in async functions and the top level bodies of modules (at app.js:5:1)`

Comment (ninadbstack) [reply packages/core/test/discovery.test.js:1996]: Oh nice, did not know this :P

**Review by ninadbstack — APPROVED** [review]

**Pull request merged**